src/runtime/opencl.cpp

```diff
@@ -720,7 +724,9 @@ WEAK int halide_opencl_device_sync(void *user_context, halide_buffer_t *) {
     debug(user_context) << "CL: halide_opencl_device_sync (user_context: " << user_context << ")\n";
 
     ClContext ctx(user_context);
-    halide_assert(user_context, ctx.error == CL_SUCCESS);
+    if (ctx.error != CL_SUCCESS) {
```

Comment:
Generally on failure we call halide_error *and* return a non-zero integer, so this should do error(user_context) << ... like below.

Reply:
Done